authority/admin/api/provisioner.go

```diff
@@ -179,9 +188,48 @@ func UpdateProvisioner(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// validate the templates and template data
+	if err := validateTemplates(nu.X509Template, nu.SshTemplate); err != nil {
+		render.Error(w, admin.WrapError(admin.ErrorBadRequestType, err, "invalid template"))
+		return
+	}
+
 	if err := auth.UpdateProvisioner(r.Context(), nu); err != nil {
 		render.Error(w, err)
 		return
 	}
 	render.ProtoJSON(w, nu)
 }
+
+// validateTemplates validates the X.509 and SSH templates and template data if set.
+func validateTemplates(x509 *linkedca.Template, ssh *linkedca.Template) error {
+
+	if x509 != nil {
+		if len(x509.Template) > 0 {
+			if err := x509util.ValidateTemplate(string(x509.Template)); err != nil {
+				return fmt.Errorf("invalid X.509 template: %w", err)
+			}
+		}
```

Comment:
Validating the template and failing forces the user to create more complicated templates.